### ISSUE TITLE
App showing previously visited Components 

### ISSUE THREAD

**Issue by bfheythem** [creativetimofficial/ct-material-dashboard-pro-angular#237]
I suddenly started facing this problem ; where I keep getting the previously visited pages in the main display page 
for example when I visit the Dashboard then go to another page I still see it there 
preview : 
![pbm](https://user-images.githubusercontent.com/22716581/38877858-81882bec-4257-11e8-9acb-5d96f34baac5.gif)

I don't what happened thing were doin just fine before

**Comment by chelaruc** [comment]
@bfheythem sorry for the late response. Can you update me on the progress?

**Comment by bfheythem** [comment]
@chelaruc still nthg I still dont know if it's because of the service I implemented or relating to the routing logic used in the app 
anyway this is my service code (the one with the subscribe call ) : 
```
import {Injectable} from '@angular/core';
import {Http, Response} from '@angular/http';
import {Observable} from 'rxjs/Observable';
import {baseURL} from '../../../shared/baseurl';
import {of} from 'rxjs/observable/of';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {BillModel} from '../models/Bill.model';
import {catchError, map, tap} from 'rxjs/operators';
import 'rxjs/Rx';

const httpOptions = {
    headers: new HttpHeaders({'Content-Type': 'application/json'})
};

@Injectable()
export class BillService {

    constructor(private http: HttpClient) {
    }

    /* Uses http.get() to load data from a single API endpoint */
    getBills() {
        return this.http.get<BillModel>('http://localhost:5000/bills/123')
            .map((response) => {
                return response['result'][0];
            } )
            .pipe(
                catchError(this.handleError('getBills', []))
            );
    }

    private handleError<T> (operation = 'operation', result?: T) {
        return (error: any): Observable<T> => {

            // TODO: send the error to remote logging infrastructure
            console.error(error); // log to console instead


            // Let the app keep running by returning an empty result.
            return of(result as T);
        };
    }
}
```
and this is the code of the new Module I created : 
```
@NgModule({
    imports: [
        CommonModule,
        RouterModule.forChild(BillsRoutes),
        FormsModule,
        MaterialModule,
        HttpClientModule,

    ],
    declarations: [BillsComponent, AddNewBillComponent, DataTableComponent]
    ,
    providers: [ BillService],
})
export class BillsModule {
}
```
and this is routing of that module : 

```
export const BillsRoutes: Routes = [
    {
        path: '',
        /*children: [ {
            path: '',*/
            component: BillsComponent
       // }]
    }];
```

PS: I made some changes here passing the component directly without "childeren [ ]" 

and this is the app.routing module : 

```
{
        path: '',
        component: AdminLayoutComponent,
        children: [
            {
                path: '',
                loadChildren: './dashboard/dashboard.module#DashboardModule'
            }, {
                path: 'components',
                loadChildren: './components/components.module#ComponentsModule'
            },  {
                path: 'charts',
                loadChildren: './charts/charts.module#ChartsModule'
            },
                {
                     path: 'bills',
                     loadChildren: './Bills/bills.module#BillsModule'
                 }
/*
            { path: 'bills', component: BillsComponent},
*/
        ]
    }
```

I hope the pbm desc is clear

**Comment by chelaruc** [comment]
@bfheythem have you got errors on console? Why you change this?
```
export const BillsRoutes: Routes = [
    {
        path: '',
        /*children: [ {
            path: '',*/
            component: BillsComponent
       // }]
    }];
```

**Comment by bfheythem** [comment]
@chelaruc 
I posted a question on stackoverflow and some one suggested it , but yet nthg has been solved : 
I recently identified the cause of the pbm : 
1/ when I try to display the incoming data from the web service using "*ngfor +<p> tag"  the problem does not appear 
however when I pass that data directly to tthe "tabledata.datarow" inside the "subscribe " , the problem starts occurring and I get component showing on top of each other 

so basically there is something wrong with the data passing to the table throw async mechanism or smtgh

**Comment by bfheythem** [comment]
![image](https://user-images.githubusercontent.com/22716581/39297710-2100052e-493c-11e8-96b9-380dee69537c.png)

2/ Second thing I noticed is that when I navigate from component to another every thing works prefectly but when I go to and external link and the go back to the page where the data should be displayed the problem starts occuring again

**Comment by chelaruc** [comment]
@bfheythem have you tried with a setTimeOut function?
